week2/homework/src/index.js

```diff
@@ -1,3 +1,70 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+
+const fs = require('fs');
+
+let fileName = 'toDo.json';
+// console.log(fileName);
+function readFile() {
+    return new Promise(
+        resolve => fs.readFile(
+            fileName,
+            (err, data) => resolve(err ? '' : data.toString())
+        )
+    );
+}
+
+function remove() {
+    readFile()
+        .then(content => {
+            let newArray = content.split('\n');
+            let i = newArray.length-1
+            newArray.splice(i, 1);
+           let new2 =  newArray.join('\n')    
+            writeFile(new2);
+            console.log(content);
+        })
+}
+
+function writeFile(...text) {
+    return new Promise(
+        (resolve, reject) => fs.appendFile(
```

Comment:
You are assuming that the file exists here. What happens if you try to `appendFile` and there is no `toDo.json`?